MH4U Dex/CoreDataController.m

```diff
@@ -363,6 +438,87 @@ - (NSUInteger)countForEntityWithEntityName:(NSString *)entityName withPredicate:
     }
 }
 
+- (QuestDrop *)questDropFromDictionary:(NSDictionary *)dropDict
+{
+    NSPredicate *questPredicate = [NSPredicate predicateWithFormat:@"%K == %@", QuestAttributes.id, dropDict[QuestDropQuestIDKey]];
+    if (![self getUniqueEntityWithEntityName:[Quest entityName] withPredicate:questPredicate]) {
```

Comment:
To avoid having to run this `getUniqueEntityWithEntityName:withPredicate:` call twice, you could store it into a local variable, e.g.:
```objc
Quest *quest = (Quest *)[self getUniqueEntityWithEntityName:[Quest entityName] withPredicate:questPredicate];
if (!quest) {
...
drop.quest = quest.
```